Running "make oct" fails

OS: Fedora Silverblue 28
CPU: AMD® Ryzen 5 1600 six-core processor × 12 
GPU: Radeon RX 580 Series (POLARIS10 / DRM 3.26.0 / 4.18.8-200.fc28.x86_64, LLVM 6.0.1)
With GNU Octave 4.4.1

See: https://github.com/flathub/org.octave.Octave/issues/33

From the provided link it seems like there is an error making mmc.mex with octave 4.4.1 when running 
`
system(make oct) 
`
From within Octave.

```
Building bin/mmc
CC=cc CXX=g++ LFLAGS='-fopenmp' LDFLAGS='' CPPFLAGS='-c -Wall -g -fno-strict-aliasing -m64 -DMMC_USE_SSE -DHAVE_SSE2 -msse -msse2 -msse3 -mssse3 -msse4.1 -O3 -fopenmp  -fPIC -DMCX_CONTAINER -DUSE_OS_TIMER -std=c++11'  mkoctfile  --mex mmclab.cpp -I../src -o bin/mmc built/posix_randr.o built/simpmesh.o built/tettracing.o built/mcx_utils.o built/tictoc.o built/mmc.o built/mmc_host.o built/cjson/cJSON.o -o ../mmclab/mmc
mmclab.cpp: In function ‘void mexFunction(int, mxArray**, int, constmxArray**)’:
mmclab.cpp:273:95: error: cannot convert ‘dimtype* {aka int*}’ to ‘const mwSize* {aka const long int*}’ for argument ‘2’ to ‘mxArray* mxCreateNumericArray(mwSize, const mwSize*, mxClassID, mxComplexity)’
 ,jstruct,0, mxCreateNumericArray(3,fielddim,mxSINGLE_CLASS,mxREAL));
                                                                  ^
mmclab.cpp:279:103: error: cannot convert ‘dimtype* {aka int*}’ to ‘const mwSize* {aka const long int*}’ for argument ‘2’ to ‘mxArray* mxCreateNumericArray(mwSize, const mwSize*, mxClassID, mxComplexity)’
 ,jstruct,0, mxCreateNumericArray(2,fielddim,mxSINGLE_CLASS,mxREAL));
                                                                  ^
mmclab.cpp:284:115: error: cannot convert ‘dimtype* {aka int*}’ to ‘const mwSize* {aka const long int*}’ for argument ‘2’ to ‘mxArray* mxCreateNumericArray(mwSize, const mwSize*, mxClassID, mxComplexity)’
 ],jstruct,0, mxCreateNumericArray(2,fielddim,mxUINT8_CLASS,mxREAL));
                                                                  ^
mmclab.cpp:350:101: error: cannot convert ‘dimtype* {aka int*}’ to ‘const mwSize* {aka const long int*}’ for argument ‘2’ to ‘mxArray* mxCreateNumericArray(mwSize, const mwSize*, mxClassID, mxComplexity)’
 ,jstruct,0, mxCreateNumericArray(4,fielddim,mxDOUBLE_CLASS,mxREAL));
                                                                  ^
mmclab.cpp:352:105: error: cannot convert ‘dimtype* {aka int*}’ to ‘const mwSize* {aka const long int*}’ for argument ‘2’ to ‘mxArray* mxCreateNumericArray(mwSize, const mwSize*, mxClassID, mxComplexity)’
 ,jstruct,0, mxCreateNumericArray(2,fielddim,mxDOUBLE_CLASS,mxREAL));
                                                                  ^
mmclab.cpp: In function ‘void mmc_set_field(const mxArray*, const mxArray*, int, mcconfig*, tetmesh*)’:
mmclab.cpp:438:38: error: cannot convert ‘const mwSize* {aka const long int*}’ to ‘const dimtype* {aka const int*}’ in assignment
         arraydim=mxGetDimensions(item);
                                      ^
mmclab.cpp:450:38: error: cannot convert ‘const mwSize* {aka const long int*}’ to ‘const dimtype* {aka const int*}’ in assignment
         arraydim=mxGetDimensions(item);
                                      ^
mmclab.cpp:463:38: error: cannot convert ‘const mwSize* {aka const long int*}’ to ‘const dimtype* {aka const int*}’ in assignment
         arraydim=mxGetDimensions(item);
                                      ^
mmclab.cpp:474:38: error: cannot convert ‘const mwSize* {aka const long int*}’ to ‘const dimtype* {aka const int*}’ in assignment
         arraydim=mxGetDimensions(item);
                                      ^
mmclab.cpp:487:38: error: cannot convert ‘const mwSize* {aka const long int*}’ to ‘const dimtype* {aka const int*}’ in assignment
         arraydim=mxGetDimensions(item);
                                      ^
mmclab.cpp:498:38: error: cannot convert ‘const mwSize* {aka const long int*}’ to ‘const dimtype* {aka const int*}’ in assignment
         arraydim=mxGetDimensions(item);
                                      ^
mmclab.cpp:510:38: error: cannot convert ‘const mwSize* {aka const long int*}’ to ‘const dimtype* {aka const int*}’ in assignment
         arraydim=mxGetDimensions(item);
                                      ^
mmclab.cpp:565:38: error: cannot convert ‘const mwSize* {aka const long int*}’ to ‘const dimtype* {aka const int*}’ in assignment
         arraydim=mxGetDimensions(item);
                                      ^
mmclab.cpp:613:38: error: cannot convert ‘const mwSize* {aka const long int*}’ to ‘const dimtype* {aka const int*}’ in assignment
         arraydim=mxGetDimensions(item);
                                      ^
mmclab.cpp:630:38: error: cannot convert ‘const mwSize* {aka const long int*}’ to ‘const dimtype* {aka const int*}’ in assignment
         arraydim=mxGetDimensions(item);
                                      ^
mmclab.cpp:638:31: error: cannot convert ‘const mwSize* {aka const long int*}’ to ‘const dimtype* {aka const int*}’ in assignment
  arraydim=mxGetDimensions(item);
                               ^
make: *** [../commons/Makefile_common.mk:180: bin/mmc] Error 1
ans =  2
```

--- a/src/mmclab.cpp
+++ b/src/mmclab.cpp
@@ -63,11 +63,7 @@
 #define MAX(a,b)  ((a)>(b)?(a):(b))                       //! Macro to calculate the max of two floating points
 #define MEXERROR(a)  mcx_error(999,a,__FILE__,__LINE__)   //! Macro to add unit name and line number in error printing
 
-#if (! defined MX_API_VER) || (MX_API_VER < 0x07300000)
-	typedef int dimtype;                              //! MATLAB before 2017 uses int as the dimension array
-#else
-	typedef size_t dimtype;                           //! MATLAB after 2017 uses size_t as the dimension array
-#endif
+typedef mwSize dimtype;                                   //! MATLAB type alias for integer type to use for array sizes and dimensions
 
 void mmc_set_field(const mxArray *root,const mxArray *item,int idx, mcconfig *cfg, tetmesh *mesh);
 void mmc_validate_config(mcconfig *cfg, tetmesh *mesh);

